test/suite/test_hs25.py

```diff
@@ -0,0 +1,78 @@
+import wttest, wiredtiger
+from wtdataset import SimpleDataSet
+
+def timestamp_str(t):
+    return '%x' % t
+
+# test_hs25.py
+# Ensure updates structure is correct when processing each key.
+class test_hs25(wttest.WiredTigerTestCase):
+    def large_updates(self, uri, value, nrows, commit_ts):
+        # Update a large number of records, we'll hang if the history store table isn't working.
+        cursor = self.session.open_cursor(uri)
+        for i in range(1, nrows + 1):
+            self.session.begin_transaction()
+            cursor.set_key(str(i))
+            cursor.set_value(str(i) + value)
+            self.assertEqual(cursor.update(), 0)
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(commit_ts))
+        cursor.close()
+
+    def prepare_updates(self, uri, value, nrows, prepare_ts):
```

Comment:
I won't do a prepared update on each key. Instead I would do a prepared update every two updates.

Reply:
The problem only triggers with a non prepared key follows a prepared key.

Reply:
Also no need to use a lot of keys. I think two keys is enough to reproduce this.